website/docs/client/java/partials/java_dynamic_env_vars.md

```diff
@@ -0,0 +1,143 @@
+
+```sh
+OPENLINEAGE__TRANSPORT__TYPE=http
+OPENLINEAGE__TRANSPORT__URL=http://localhost:5050
+OPENLINEAGE__TRANSPORT__ENDPOINT=/api/v1/lineage
+OPENLINEAGE__TRANSPORT__AUTH__TYPE=api_key
+OPENLINEAGE__TRANSPORT__AUTH__API_KEY=random_token
+OPENLINEAGE__TRANSPORT__AUTH__COMPRESSION=gzip
+```
+
+is equivalent to passing following YAML configuration:
+```yaml
+transport:
+  type: http
+  url: http://localhost:5050
+  endpoint: api/v1/lineage
+  auth:
+  type: api_key
+    apiKey: random_token
+    compression: gzip
+```
+</TabItem>
+
+<TabItem value="composite" label="Composite Example">
+
+Setting following environment variables:
+
+```sh
+OPENLINEAGE__TRANSPORT__TYPE=composite
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__TYPE=http
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__TYPE=http
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__URL=http://localhost:5050
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__ENDPOINT=/api/v1/lineage
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__AUTH__TYPE=api_key
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__AUTH__API_KEY=random_token
+OPENLINEAGE__TRANSPORT__TRANSPORTS__FIRST__AUTH__COMPRESSION=gzip
+OPENLINEAGE__TRANSPORT__TRANSPORTS__SECOND__TYPE=console
+```
+
+is equivalent to passing following YAML configuration:
+```yaml
+transport:
+  type: composite
+  transports:
+    first:
+      type: http
+      url: http://localhost:5050
+      endpoint: api/v1/lineage
+      auth:
+        type: api_key
+        apiKey: random_token
+      compression: gzip
+    second:
+      type: console
+```
+</TabItem>
+
+<TabItem value="precedence" label="Precedence Example">
+
+Setting following environment variables:
+
+```sh
+OPENLINEAGE__TRANSPORT='{"type":"console"}'
+OPENLINEAGE__TRANSPORT__TYPE=http
+```
+
+is equivalent to passing following YAML configuration:
+```yaml
+transport:
+  type: console
+```
+</TabItem>
+
+<TabItem value="spark-properties" label="Spark Example">
+
+Setting following environment variables:
+
+```sh
+OPENLINEAGE__TRANSPORT__TYPE=kafka
+OPENLINEAGE__TRANSPORT__TOPIC_NAME=test
+OPENLINEAGE__TRANSPORT__MESSAGE_KEY=explicit-key
+OPENLINEAGE__TRANSPORT__PROPERTIES='{"key.serializer": "org.apache.kafka.common.serialization.StringSerializer"}'
+```
+
+is equivalent to passing following YAML configuration:
+```yaml
+transport:
+  type: kafka
+  topicName: test
+  messageKey: explicit-key
+  properties:
+    key.serializer: org.apache.kafka.common.serialization.StringSerializer
+```
+
+Please note that you can't use environment variables to set Spark properties, as they are not part of the configuration hierarchy.
+Following environment variable:
+```sh
+OPENLINEAGE__TRANSPORT__PROPERTIES__KEY__SERIALIZER="org.apache.kafka.common.serialization.StringSerializer"
+```
+would be equivalent to below YAML structure:
+```yaml
+transport:
+  properties:
+    key:
+      serializer: org.apache.kafka.common.serialization.StringSerializer
+```
+which is not a valid configuration for Spark.
+
+</TabItem>
+
```

Comment:
My understanding is that env vars do not allow providing array params.
Perhaps this may be documented.

Reply:
For example https://openlineage.io/docs/client/java/configuration#host-list-resolver -> has an ability to provide a list of hosts. Is this doable to express though env vars? I thought no. If no. it may be worth to put this into docs.

Reply:
I've added another example in docs as well as unit test that covers this.

One thing that could be mentioned explicitly is that one can't use `-` in env vars. This means e.g. for namespaceResolvers you may name them as follows in the test: `resolvedName` but not `resolved-name`. Not sure if those names matter at all but maybe it's worth to mention.